spec/system/admin/enterprise_fees_spec.rb

```diff
@@ -177,27 +177,56 @@
       login_as enterprise_user
     end
 
-    it "creates enterprise fees" do
-      visit edit_admin_enterprise_path(distributor1)
-      within(".side_menu") { click_link 'Enterprise Fees' }
-      click_link "Create One Now"
-
-      select distributor1.name,
-             from: 'sets_enterprise_fee_set_collection_attributes_0_enterprise_id'
-      select 'Packing', from: 'sets_enterprise_fee_set_collection_attributes_0_fee_type'
-      fill_in 'sets_enterprise_fee_set_collection_attributes_0_name', with: 'foo'
-      select 'GST', from: 'sets_enterprise_fee_set_collection_attributes_0_tax_category_id'
-      select 'Flat Percent', from: 'sets_enterprise_fee_set_collection_attributes_0_calculator_type'
-      click_button 'Update'
-
-      expect(flash_message).to eq('Your enterprise fees have been updated.')
-
-      # After saving, we should be redirected to the fees for our chosen enterprise
-      expect(page).not_to have_select 'sets_enterprise_fee_set_collection_attributes_1_enterprise_id',
-                                      selected: 'Second Distributor'
-
-      enterprise_fee = EnterpriseFee.find_by name: 'foo'
-      expect(enterprise_fee.enterprise).to eq(distributor1)
+    context "creating an enterprise fee" do
+      before do
+        visit edit_admin_enterprise_path(distributor1)
+        within(".side_menu") { click_link 'Enterprise Fees' }
+        click_link "Create One Now"
+      end
+
+      shared_examples "shared example" do |tax_category, calculator, flash_message, fee_count|
+        context "setting it up as #{tax_category}, with a #{calculator} calculator" do
+          it "triggers the expected message" do
+            select distributor1.name,
+                   from: 'sets_enterprise_fee_set_collection_attributes_0_enterprise_id'
+            select 'Packing', from: 'sets_enterprise_fee_set_collection_attributes_0_fee_type'
+            fill_in 'sets_enterprise_fee_set_collection_attributes_0_name', with: 'foo'
+            select tax_category, from: 'sets_enterprise_fee_set_collection_attributes_0_tax_category_id'
+            select calculator, from: 'sets_enterprise_fee_set_collection_attributes_0_calculator_type'
+            click_button 'Update'
+
+            # The correct flash message should be displayed
+            expect(page).to have_content(flash_message)
+
+            # After saving, we should be redirected to the fees for our chosen enterprise
+            expect(page).not_to have_select 'sets_enterprise_fee_set_collection_attributes_1_enterprise_id',
+                                            selected: 'Second Distributor'
+            # A new enterprise fee is created
+            expect(EnterpriseFee.count).to eq(fee_count)
+          end
+        end
+      end
+
+      xcontext "an error message is displayed" do
+        # pending "#10348"
```

Comment:
Why `x` instead of `pending`?

Reply:
I could not get it to work with pending, I think this is because pending would need to be inside the `shared_examples` block, which would influence the three other (not pending) test cases.

So, If I remove the `x`, and uncomment the pending line, the spec fails. I thought of adding a condition within the `shared_examples` but it seems like too much hassle...

Any ideas on how to make it work with pending in this case?

Reply:
Ah, now I see. All good. Shared examples with some pending is not easy to write. You found the simplest solution.